internal/service/meta/meta_service.go

```diff
@@ -92,3 +99,59 @@ func (ms *MetaService) GetMetaList(ctx context.Context, objID string) (metas []*
 	}
 	return metas, err
 }
+
+// GetReactionByObjectId get reaction
+func (ms *MetaService) GetReactionByObjectId(ctx context.Context, objectID string) (resp schema.ReactionResp, err error) {
+	reactionMeta, err := ms.GetMetaByObjectIdAndKey(ctx, objectID, entity.ObjectReactSummaryKey)
+
+	// if not exist, return nil
+	if err != nil {
+		var pacmanErr *myErrors.Error
+		if errors.As(err, &pacmanErr) && pacmanErr.Reason == reason.MetaObjectNotFound {
+			return nil, nil
+		} else {
+			return nil, err
+		}
+	}
+
+	var reaction schema.ReactSummaryMeta
+	err = json.Unmarshal([]byte(reactionMeta.Value), &reaction)
+	if err != nil {
+		return nil, err
+	}
+	return ms.convertToReactionResp(ctx, reaction)
+}
+
+// AddOrUpdateReaction add or update reaction
+func (ms *MetaService) AddOrUpdateReaction(ctx context.Context, req *schema.UpdateReactionReq) (resp schema.ReactionResp, err error) {
+	reaction, err := ms.metaRepo.AddOrUpdateMetaByObjectIdAndKey(ctx, req)
```

Comment:
Hi @LinkinStars , I plan to use `QuestionRepo.GetQuestion` to check if there's any existing question in `AddOrUpdateReaction`, but got "import cycle not allowed".
```
        ...
        imports github.com/apache/incubator-answer/internal/service/object_info
	imports github.com/apache/incubator-answer/internal/service/question_common
	imports github.com/apache/incubator-answer/internal/service/meta
	imports github.com/apache/incubator-answer/internal/service/question_common: import cycle not allowed
```

I found that's becasue `question.go`, where `QuestionRepo` lives in, also imports `../internal/service/meta`. That causes this issue.
Because the `QuestionRepo` directly talks to DB and I can't find a work around. Any suggestion for this? thx!

Reply:
Wow, you're really responsive even so late! Thanks for your suggestions and I'll take a look tomorrow.

Reply:
updated based on your suggestion.